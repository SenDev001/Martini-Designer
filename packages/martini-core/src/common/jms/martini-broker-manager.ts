export const martiniBrokerPath = "/services/martini/broker";
export const destinationPattern = /^(queue|topic):\/\/([\w\d\W]+)$/;

export interface Destination {
    type: DestinationType;
    name: string;
}

export namespace Destination {
    export function getType(destination: string): DestinationType {
        if (destination.startsWith("queue://")) return DestinationType.QUEUE;
        else if (destination.startsWith("topic://")) return DestinationType.TOPIC;
        throw Error("Invalid destination syntax, valid syntax is " + destinationPattern);
    }

    export function getName(destination: string): string {
        return destination.replace(/^(topic|queue):\/\//, "");
    }

    export function toString(destination: Destination): string {
        return `${destination.type.toLowerCase()}://${destination.name}`;
    }

    export function toDestination(destinationString: string): Destination | undefined {
        try {
            return {
                type: Destination.getType(destinationString),
                name: Destination.getName(destinationString)
            };
        } catch (error) {
            return undefined;
        }
    }

    export function equals(destination: Destination, other: Destination) {
        return destination.name === other.name && destination.type === other.type;
    }
}

export enum DestinationType {
    QUEUE = "QUEUE",
    TOPIC = "TOPIC"
}

export const MartiniBrokerManager = Symbol("MartiniBrokerManager");

export interface MartiniBrokerManager {
    getDestinations(): Promise<Destination[]>;

    sendString(destination: string, message: string): Promise<void>;

    sendBytes(destination: string, message: string): Promise<void>;

    sendBytesFromUrl(destination: string, url: string): Promise<void>;
}
