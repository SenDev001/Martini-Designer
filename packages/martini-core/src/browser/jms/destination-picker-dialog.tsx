import { DialogMode } from "@theia/core/lib/browser";
import { Field, Form, Formik, FormikErrors, FormikProps, yupToFormErrors } from "formik";
import { inject, injectable, interfaces } from "inversify";
import { capitalize } from "lodash";
import messages from "martini-messages/lib/messages";
import * as React from "react";
import * as yup from "yup";
import { Destination, DestinationType, MartiniBrokerManager } from "../../common/jms/martini-broker-manager";
import { List, ListItem } from "../components/list";
import { RadioButton } from "../components/radio-button";
import { applySize, BaseDialog } from "../dialogs/dialogs";
import { FormRow, OnFormChange } from "../form/form";
import { Progress } from "../progress/progress-service";

export interface DestinationListProps {
    defaultSelectedDestination?: Destination;
    destinations: Destination[];
    onSelectionChange?: (destination: Destination) => void;
    onDoubleClick?: (destination: Destination) => void;
}

export const DestinationList: React.FC<DestinationListProps> = ({ defaultSelectedDestination, destinations, onSelectionChange, onDoubleClick }) => {
    const destinationItems: ListItem[] = React.useMemo(() => destinations
        .map(destination => ({
            label: Destination.toString(destination),
            data: destination,
            selected: defaultSelectedDestination && Destination.equals(destination, defaultSelectedDestination),
            iconClass: `martini-tree-icon martini-jms-${destination.type.toLowerCase()}-icon`
        })), [destinations]);

    const handleSelectionChange = (item: ListItem) => {
        if (onSelectionChange)
            onSelectionChange(item.data);
    };

    const handleDoubleClick = (item: ListItem) => {
        if (onDoubleClick)
            onDoubleClick(item.data);
    };

    return <List
        focus={true}
        filtered={true}
        items={destinationItems}
        onSelectionChanged={handleSelectionChange}
        onDoubleClick={handleDoubleClick}
        style={{ height: "100%" }}
    />;
};

export interface DestinationPickerProps {
    destinations: Destination[];
    selectedDestination?: Destination;
    onChange: (destination: Destination | undefined) => void;
    onValidate?: (destination: Destination | undefined, errors: FormikErrors<DestinationConfig>) => void;
    onSelect?: (destination: Destination) => void;
}

interface DestinationConfig {
    create: boolean;
    selectedDestination?: Destination;
    newDestinationName: string;
    newDestinationType: DestinationType;
}

export const DestinationPicker: React.FC<DestinationPickerProps> = (
    {
        destinations,
        selectedDestination,
        onChange,
        onValidate,
        onSelect,
    }
) => {
    const schema = React.useMemo(() => yup.object()
        .shape({
            newDestinationName: yup.string()
                .when("create", {
                    is: true,
                    then: yup.string()
                        .trim()
                        .required()
                        .test('unique',
                            messages.destination_name_exists,
                            value => !destinations.map(destination => destination.name).some(name => name === value))
                })
        }), [destinations]);

    const handleFormChange = (config: DestinationConfig) => {
        if (config.create) {
            onChange({
                name: config.newDestinationName,
                type: config.newDestinationType
            });
        } else
            onChange(config.selectedDestination);
    };

    return <Formik
        initialValues={{
            create: false,
            selectedDestination,
            newDestinationName: "",
            newDestinationType: DestinationType.QUEUE
        }}
        initialTouched={{
            create: true,
            selectedDestination: true,
            newDestinationName: true,
            newDestinationType: true
        }}
        validate={async (config: DestinationConfig) => {
            let errors: FormikErrors<DestinationConfig> = {};

            try {
                await schema.validate(config, { abortEarly: false });
            } catch (error) {
                errors = yupToFormErrors(error);
            }

            if (onValidate)
                onValidate(config.create ?
                    { name: config.newDestinationName, type: config.newDestinationType } :
                    config.selectedDestination, errors);

            return errors;
        }}
        enableReinitialize={true}
        validateOnMount={true}
        onSubmit={_ => {
        }}
    >
        {(props: FormikProps<DestinationConfig>) => {
            const { values, setValues } = props;
            const handleCreateChange = (create: boolean) => {
                setValues({
                    ...values,
                    create
                });
            };

            const handleSelectedDestinationChange = (selectedDestination: Destination) => {
                setValues({
                    ...values,
                    selectedDestination
                });
            };

            const handleDestinationDoubleClick = (selectedDestination: Destination) => {
                if (onSelect)
                    onSelect(selectedDestination);
            };

            const handleTypeChange = (newDestinationType: DestinationType) => {
                setValues({
                    ...values,
                    newDestinationType
                });
            };

            return <Form
                style={{
                    display: "grid",
                    gridColumnGap: 10,
                    gridTemplateColumns: "max-content 1fr"
                }}
            >
                <OnFormChange onChange={handleFormChange} />
                <FormRow name="create" label="From:">
                    <RadioButton
                        name="create"
                        value={false}
                        label={messages.existing_destinations_title}
                        onChange={_ => handleCreateChange(false)}
                        checked={!values.create}
                    />
                    <RadioButton
                        name="create"
                        label={messages.new_destinations_title}
                        value={true}
                        onChange={_ => handleCreateChange(true)}
                        checked={values.create}
                    />
                </FormRow>

                {!values.create ? (
                    <div style={{ gridColumn: "span 2", height: "310px" }}>
                        <DestinationList
                            onSelectionChange={handleSelectedDestinationChange}
                            defaultSelectedDestination={values.selectedDestination}
                            destinations={destinations}
                            onDoubleClick={handleDestinationDoubleClick}
                        />
                    </div>
                ) : (
                        <>
                            <FormRow name="newDestinationName" label="Name:">
                                <Field
                                    name="newDestinationName"
                                    className="theia-input"
                                />
                            </FormRow>
                            <FormRow name="newDestinationType" label="Type:">
                                <RadioButton
                                    name="newDestinationType"
                                    label={capitalize(DestinationType.QUEUE)}
                                    value={DestinationType.QUEUE}
                                    onChange={_ => handleTypeChange(DestinationType.QUEUE)}
                                    checked={values.newDestinationType === DestinationType.QUEUE}
                                />
                                <RadioButton
                                    name="newDestinationType"
                                    label={capitalize(DestinationType.TOPIC)}
                                    value={DestinationType.TOPIC}
                                    onChange={_ => handleTypeChange(DestinationType.TOPIC)}
                                    checked={values.newDestinationType === DestinationType.TOPIC}
                                />
                            </FormRow>
                        </>
                    )}
            </Form>;
        }}
    </Formik>;
};

export interface DestinationPickerDialogProps {
    initialDestination?: Destination;
}

export const DestinationPickerDialogProps = Symbol("DestinationPickerDialogProps");

@injectable()
export class DestinationPickerDialog extends BaseDialog<Destination> {
    @inject(MartiniBrokerManager)
    private readonly brokerManager: MartiniBrokerManager;

    private destination: Destination;
    private destinations: Destination[] = [];
    private valid = true;

    constructor(
        @inject(DestinationPickerDialogProps)
        private readonly _props: DestinationPickerDialogProps
    ) {
        super({
            title: messages.destination_dialog_title
        });
        applySize(this.contentNode, { width: 500, height: 400 });
        this.appendCloseButton(messages.cancel_btn);
        this.appendAcceptButton(messages.apply_btn);
        this.destination = _props.initialDestination!;
    }

    protected async init(progress: Progress): Promise<void> {
        progress.report({
            message: messages.fetching_jms_destinations
        });
        this.destinations = (await this.brokerManager.getDestinations());
        return super.init(progress);
    }

    protected isValid(value: Destination, mode: DialogMode) {
        return this.valid;
    }

    get value(): Destination {
        return this.destination;
    }

    protected doRender(): React.ReactNode {
        return <DestinationPicker
            selectedDestination={this._props.initialDestination}
            onChange={destination => this.handleChange(destination)}
            onValidate={(_, errors) => this.handleValidate(_, errors)}
            onSelect={async destination => {
                this.handleChange(destination);
                await this.accept();
            }}
            destinations={this.destinations} />;
    }

    private handleChange(destination: Destination | undefined) {
        this.destination = destination!;
    }

    private handleValidate(destination: Destination | undefined, errors: FormikErrors<DestinationConfig>) {
        this.valid = Object.keys(errors).length === 0 && destination !== undefined;
        this.validate();
    }
}

export const bindDestinationPickerDialogBindings = (bind: interfaces.Bind) => {
    bind("Factory<DestinationPickerDialog>").toFactory(ctx => (props: DestinationPickerDialogProps) => {
        const child = ctx.container.createChild();
        child.bind(DestinationPickerDialogProps).toConstantValue(props);
        child.bind(DestinationPickerDialog).toSelf();
        return child.get(DestinationPickerDialog);
    });
};
