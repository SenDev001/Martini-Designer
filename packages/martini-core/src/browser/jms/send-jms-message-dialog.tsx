import { DialogMode } from "@theia/core/lib/browser";
import { encode } from "base64-arraybuffer";
import { Field, Form, Formik, FormikErrors } from "formik";
import { inject, injectable, interfaces } from "inversify";
import { noop } from "lodash";
import messages from "martini-messages/lib/messages";
import * as React from "react";
import * as yup from 'yup';
import { Destination, destinationPattern, MartiniBrokerManager } from "../../common/jms/martini-broker-manager";
import { FilePicker } from "../components/file-picker";
import { RadioButton } from "../components/radio-button";
import { applySize, BaseDialog, ConfirmDialog } from "../dialogs/dialogs";
import { FormEffect, FormErrorMessage, FormErrorMessageWrapper, FormRow, OnFormChange, validateSchema } from "../form/form";
import "../yup/yup-ext";
import { DestinationPickerDialog, DestinationPickerDialogProps } from "./destination-picker-dialog";

export interface SendJmsMessageFormProps {
    config: SendJmsMessageConfig,
    onBrowseDestination?: () => void;
    onValidate?: (errors: FormikErrors<SendJmsMessageConfig>) => void;
    onChange: (config: SendJmsMessageConfig) => void;
}

export enum MessageType {
    RAW_TEXT,
    FILE
}

export interface SendJmsMessageConfig {
    destination: string;
    messageType: MessageType;
    message: string;
    messageFileOrUrl: string | File;
}

export const SendJmsMessageForm: React.FC<SendJmsMessageFormProps> = (
    {
        onValidate,
        onChange,
        onBrowseDestination,
        config: {
            destination,
            messageType,
            message,
            messageFileOrUrl
        }
    }
) => {
    const schema = React.useMemo(() => yup.object().shape({
        destination: yup.string()
            .trim()
            .required()
            .matches(destinationPattern, messages.invalid_destination_format),
        message: yup.string()
            .when("messageType", {
                is: MessageType.RAW_TEXT,
                then: yup.string().trim().required()
            }),
        messageFileOrUrl: yup.mixed()
            .when("messageType", {
                is: MessageType.FILE,
                then: yup.string().when("messageFileOrUrl", {
                    is: (...values) => values.every(value => typeof value === "string"),
                    then: yup.string().trim().required().url()
                })
            })
    }), []);
    return <Formik
        initialValues={{
            destination,
            messageType,
            message,
            messageFileOrUrl
        }}
        initialTouched={{
            destination: true,
            messageType: true,
            message: true,
            messageFileOrUrl: true
        }}
        validate={config => validateSchema(schema, config, onValidate)}
        validateOnMount={true}
        enableReinitialize={true}
        onSubmit={noop}
    >
        {({ values, setValues }) => {
            const handleMessageTypeChange = (type: MessageType) => {
                setValues({
                    ...values,
                    messageType: type
                });
            };
            const handleMessageFileChange = (fileOrUrl: string | File) => {
                setValues({
                    ...values,
                    messageFileOrUrl: fileOrUrl
                });
            };
            const handleBrownDestinationClick = () => {
                if (onBrowseDestination)
                    onBrowseDestination();
            };
            const destinationInput = React.useRef<HTMLInputElement>(null);

            React.useEffect(() => {
                if (destinationInput.current)
                    destinationInput.current.focus();
            }, []);
            return <Form
                style={{
                    display: "grid",
                    gridColumnGap: 10,
                    gridTemplateColumns: "max-content 1fr"
                }}
            >
                <OnFormChange onChange={onChange} />
                <FormEffect deps={[destination]} />
                <FormRow
                    name="destination"
                    label={messages.destination}
                    showErrorMessage={false}
                    gridTemplateColumns="1fr max-content"
                >
                    <Field type="text" name="destination" className="theia-input" innerRef={destinationInput} />
                    <input
                        type="button"
                        className="theia-button"
                        value={messages.browse_btn}
                        onClick={handleBrownDestinationClick}
                    />
                    <FormErrorMessageWrapper>
                        <FormErrorMessage name="destination" />
                    </FormErrorMessageWrapper>
                </FormRow>
                <FormRow name="message" label={messages.message}>
                    <div style={{ marginBottom: "5px", display: "grid", gridTemplateColumns: "max-content max-content" }}>
                        <RadioButton
                            name="messageType"
                            label={messages.raw_text}
                            onChange={_ => handleMessageTypeChange(MessageType.RAW_TEXT)}
                            checked={values.messageType === MessageType.RAW_TEXT}
                            value={MessageType.RAW_TEXT}
                        />
                        <RadioButton
                            name="messageType"
                            label={messages.file_or_url}
                            onChange={_ => handleMessageTypeChange(MessageType.FILE)}
                            checked={values.messageType === MessageType.FILE}
                            value={MessageType.FILE}
                        />
                    </div>
                    {values.messageType === MessageType.RAW_TEXT ? (
                        <Field
                            style={{ resize: "none" }}
                            as="textarea"
                            name="message"
                            className="theia-input"
                            rows="5"
                        />
                    ) : (
                            <FilePicker
                                onChange={handleMessageFileChange}
                                value={values.messageFileOrUrl}
                                placeholder={messages.drop_browse_file_url}
                            >
                                <FormErrorMessageWrapper>
                                    <FormErrorMessage name="messageFileOrUrl" />
                                </FormErrorMessageWrapper>
                            </FilePicker>
                        )}
                </FormRow>
            </Form>;
        }}
    </Formik>;
};

export class SendJmsMessageDialogProps {
}

@injectable()
export class SendJmsMessageDialog extends BaseDialog<void> {
    @inject("Factory<DestinationPickerDialog>")
    private readonly destinationDialogFactory: (props: DestinationPickerDialogProps) => DestinationPickerDialog;
    @inject(MartiniBrokerManager)
    private readonly brokerManager: MartiniBrokerManager;
    private valid: boolean = false;

    private config: SendJmsMessageConfig = {
        messageType: MessageType.RAW_TEXT,
        message: "",
        destination: "",
        messageFileOrUrl: ""
    };
    value: undefined;

    constructor(
        @inject(SendJmsMessageDialogProps)
        props: SendJmsMessageDialogProps
    ) {
        super({ title: messages.send_jms_message_dialog_title });
        applySize(this.contentNode, { height: 220, width: 600 });
        this.appendCloseButton(messages.cancel_btn);
        this.appendAcceptButton(messages.send_btn);
    }

    protected isValid(value: undefined, mode: DialogMode) {
        return this.valid;
    }

    protected doRender(): React.ReactNode {
        return <>
            <SendJmsMessageForm
                config={this.config}
                onValidate={errors =>
                    this.handleValidate(errors)}
                onChange={config => this.handleChange(config)}
                onBrowseDestination={() => this.handleBrowseDestination()}
            />
        </>;
    }

    protected async accept(): Promise<void> {
        const { messageType, destination, message, messageFileOrUrl } = this.config;
        await this.showProgress(messages.sending_jms_message, async _ => {
            if (messageType === MessageType.RAW_TEXT)
                await this.brokerManager.sendString(destination, message);
            else if (messageType === MessageType.FILE) {
                if (typeof messageFileOrUrl === "string")
                    await this.brokerManager.sendBytesFromUrl(destination, messageFileOrUrl as string);
                else if (messageFileOrUrl instanceof File) {
                    const file = encode(await messageFileOrUrl.arrayBuffer());
                    await this.brokerManager.sendBytes(destination, file);
                }
            }

            new ConfirmDialog({
                title: messages.success_title,
                msg: messages.message_sent,
                showCancel: false
            }).open();
        });
    }

    private handleChange(config: SendJmsMessageConfig) {
        this.config = config;
    }

    private handleValidate(errors: FormikErrors<SendJmsMessageConfig>) {
        this.valid = Object.keys(errors).length === 0;
        this.validate();
    }

    private async handleBrowseDestination() {
        const destination = await this.destinationDialogFactory({
            initialDestination: Destination.toDestination(this.config.destination)
        }).open();
        if (destination) {
            this.config.destination = Destination.toString(destination);
            this.update();
        }
    }
}

export const bindSendJmsMessageDialogBindings = (bind: interfaces.Bind) => {
    bind("Factory<SendJmsMessageDialog>").toFactory(ctx => (props: SendJmsMessageDialogProps) => {
        const child = ctx.container.createChild();
        child.bind(SendJmsMessageDialogProps).toConstantValue(props);
        child.bind(SendJmsMessageDialog).toSelf();
        return child.get(SendJmsMessageDialog);
    });
};
