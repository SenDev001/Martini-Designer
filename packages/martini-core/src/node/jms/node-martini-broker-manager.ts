import { ConnectionHandler, JsonRpcConnectionHandler } from "@theia/core";
import { decode } from "base64-arraybuffer";
import { inject, injectable, interfaces } from "inversify";
import { UrlReader } from "../../common/fs/url-reader";
import { Destination, MartiniBrokerManager, martiniBrokerPath } from "../../common/jms/martini-broker-manager";
import { AxiosInstanceFactory } from "../http/axios-instance-factory";

@injectable()
export class MartiniBrokerManagerNode implements MartiniBrokerManager {
    @inject(UrlReader)
    private readonly urlReader: UrlReader;
    @inject(AxiosInstanceFactory)
    private readonly axiosFactory: AxiosInstanceFactory;

    async getDestinations(): Promise<Destination[]> {
        const response = await (await this.axiosFactory.make()).get("/esbapi/broker/destinations", {
            params: { size: 1000 }
        });

        if (response.status !== 200) return [];

        return response.data.items.map(
            (item: any) =>
                ({
                    type: item.type.toUpperCase(),
                    name: item.name
                } as Destination)
        );
    }

    async sendString(destination: string, message: string): Promise<void> {
        const _destination = this.getDestination(destination);
        await (await this.axiosFactory.make()).post(
            `/esbapi/broker/destinations/${_destination.type.toLowerCase()}/${_destination.name}/sendString`,
            message,
            {
                headers: {
                    "Content-Type": "text/plain"
                }
            }
        );
    }

    async sendBytes(destination: string, message: string): Promise<void> {
        await this._sendBytes(destination, decode(message));
    }

    async sendBytesFromUrl(destination: string, url: string): Promise<void> {
        const message = await this.urlReader.readAsBuffer(url);
        await this._sendBytes(destination, message);
    }

    async _sendBytes(destination: string, message: ArrayBuffer): Promise<void> {
        const _destination = this.getDestination(destination);
        await (await this.axiosFactory.make()).post(
            `/esbapi/broker/destinations/${_destination.type.toLowerCase()}/${_destination.name}/sendBytes`,
            message,
            {
                headers: {
                    "Content-Type": "application/octet-stream"
                }
            }
        );
    }

    private getDestination(destination: string): Destination {
        const _destination = Destination.toDestination(destination);
        if (!_destination) throw Error(`Invalid destination: '${destination}'`);
        return _destination;
    }
}

export const bindBrokerManager = (bind: interfaces.Bind) => {
    bind(MartiniBrokerManager)
        .to(MartiniBrokerManagerNode)
        .inSingletonScope();

    bind(ConnectionHandler)
        .toDynamicValue(
            ctx =>
                new JsonRpcConnectionHandler(martiniBrokerPath, _ =>
                    ctx.container.get<MartiniBrokerManager>(MartiniBrokerManager)
                )
        )
        .inSingletonScope();
};
