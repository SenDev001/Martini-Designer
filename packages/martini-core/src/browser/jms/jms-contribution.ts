import { Command, CommandContribution, CommandRegistry, MenuContribution, MenuModelRegistry } from "@theia/core";
import { inject, injectable } from "inversify";
import messages from "martini-messages/lib/messages";
import { MartiniMenus } from "../martini-ide-contribution";
import { MARTINI_IDE_CATEGORY } from "../navigator/martini-navigator-view-contribution";
import { SendJmsMessageDialog, SendJmsMessageDialogProps } from "./send-jms-message-dialog";

export const SendJmsMessageCommand: Command = {
    id: "martini.sendJmsMessage",
    label: messages.send_jms_message,
    iconClass: "martini-icon martini-send-jms-message-icon",
    category: MARTINI_IDE_CATEGORY
};

@injectable()
export class JmsCommandContribution implements CommandContribution {
    @inject("Factory<SendJmsMessageDialog>")
    private readonly sendJmsMessageDialog: (props: SendJmsMessageDialogProps) => SendJmsMessageDialog;

    registerCommands(commands: CommandRegistry): void {
        commands.registerCommand(SendJmsMessageCommand, {
            execute: async () => await this.sendJmsMessageDialog({}).open()
        });
    }
}

@injectable()
export class JmsMenuContribution implements MenuContribution {
    registerMenus(menus: MenuModelRegistry): void {
        menus.registerMenuAction(MartiniMenus.MARTINI, {
            commandId: SendJmsMessageCommand.id
        });
    }
}
