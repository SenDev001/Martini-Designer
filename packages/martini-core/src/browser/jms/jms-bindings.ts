import { interfaces } from "inversify";
import { bindSendJmsMessageDialogBindings } from "./send-jms-message-dialog";
import { WebSocketConnectionProvider } from "@theia/core/lib/browser";
import { MartiniBrokerManager, martiniBrokerPath } from "../../common/jms/martini-broker-manager";
import { CommandContribution } from "@theia/core";
import { JmsCommandContribution, JmsMenuContribution } from "./jms-contribution";
import { MenuContribution } from "@theia/core/lib/common/menu";
import { bindDestinationPickerDialogBindings } from "./destination-picker-dialog";

export const bindJmsBindings = (bind: interfaces.Bind) => {
    bind(MartiniBrokerManager)
        .toDynamicValue(ctx => {
            const connection = ctx.container.get(WebSocketConnectionProvider);
            return connection.createProxy(martiniBrokerPath);
        })
        .inSingletonScope();
    bindDestinationPickerDialogBindings(bind);
    bindSendJmsMessageDialogBindings(bind);

    bind(CommandContribution).to(JmsCommandContribution);
    bind(MenuContribution).to(JmsMenuContribution);
};
