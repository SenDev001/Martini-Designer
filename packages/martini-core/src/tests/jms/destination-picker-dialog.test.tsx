import * as Enzyme from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import * as React from "react";
import * as Renderer from "react-test-renderer";
import "reflect-metadata";
import { DestinationPicker, DestinationList } from "../../browser/jms/destination-picker-dialog";
import { Destination } from "../../common/jms/martini-broker-manager";

Enzyme.configure({ adapter: new Adapter() });

test("DestinationPicker should be rendered", () => {
    const component = Renderer.create(<DestinationPicker
        destinations={
            [
                "queue://helloWorld",
                "topic://testing",
                "queue://nice"
            ].map(dest => Destination.toDestination(dest)!)
        }
        onChange={() => { }}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("DestinationPicker should be rendered with default selected destination", () => {
    const component = Renderer.create(<DestinationPicker
        destinations={
            [
                "queue://helloWorld",
                "topic://testing",
                "queue://nice"
            ].map(dest => Destination.toDestination(dest)!)
        }
        selectedDestination={Destination.toDestination("topic://testing")}
        onChange={() => { }}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

describe("When selecting destination source", () => {
    let wrapper: Enzyme.ReactWrapper;

    beforeEach(async () => {
        wrapper = Enzyme.mount(<DestinationPicker
            destinations={
                [
                    "queue://helloWorld",
                    "topic://testing",
                    "queue://nice"
                ].map(dest => Destination.toDestination(dest)!)
            }
            onChange={() => { }}
        />);
    });

    afterEach(() => {
        wrapper.unmount();
    });

    test("Should display destination list when existing source is checked by default", () => {
        expect(wrapper.find(DestinationList).exists()).toBe(true);
    });

    test("Should display the destination name and type when new source is checked", async () => {
        const newRadioBtn = wrapper.find("input[type='radio'][name='create'][value=true]").first();
        expect(newRadioBtn.length).toBe(1);

        expect(wrapper.find(DestinationList).exists()).toBe(true);

        newRadioBtn.simulate("change", {
            target: { checked: true }
        });

        expect(wrapper.find(DestinationList).exists()).toBe(false);
    });
});
