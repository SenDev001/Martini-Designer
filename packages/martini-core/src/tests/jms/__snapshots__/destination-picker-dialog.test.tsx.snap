// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationPicker should be rendered 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
  style={
    Object {
      "display": "grid",
      "gridColumnGap": 10,
      "gridTemplateColumns": "max-content 1fr",
    }
  }
>
  <label
    htmlFor="create"
    onContextMenu={[Function]}
    style={
      Object {
        "marginTop": "3px",
        "verticalAlign": "center",
      }
    }
  >
    From:
  </label>
  <div
    style={
      Object {
        "display": "grid",
        "gridTemplateColumns": "1fr",
        "gridTemplateRows": "max-content 1fr",
      }
    }
  >
    <div
      className="sc-AxirZ fiReNW"
    >
      <input
        checked={true}
        name="create"
        onChange={[Function]}
        type="radio"
        value={false}
      />
      <div
        className="label"
        onClick={[Function]}
      >
        Existing JMS Queues and Topics
      </div>
    </div>
    <div
      className="sc-AxirZ fiReNW"
    >
      <input
        checked={false}
        name="create"
        onChange={[Function]}
        type="radio"
        value={true}
      />
      <div
        className="label"
        onClick={[Function]}
      >
        New JMS Queue or Topic
      </div>
    </div>
    <div
      className="form-error-message-wrapper"
    />
  </div>
  <div
    style={
      Object {
        "gridColumn": "span 2",
        "height": "310px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "grid",
          "gridRowGap": "4px",
          "gridTemplateColumns": "1fr",
          "gridTemplateRows": "max-content 1fr",
          "height": "100%",
        }
      }
    >
      <input
        className="theia-input"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search..."
        type="text"
      />
      <ul
        className="sc-AxjAm hGTuJe theia-input"
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <li
          className="item "
          onClick={[Function]}
          onDoubleClick={[Function]}
        >
          <span
            className="martini-tree-icon martini-jms-queue-icon"
          />
          queue://helloWorld
        </li>
        <li
          className="item "
          onClick={[Function]}
          onDoubleClick={[Function]}
        >
          <span
            className="martini-tree-icon martini-jms-topic-icon"
          />
          topic://testing
        </li>
        <li
          className="item "
          onClick={[Function]}
          onDoubleClick={[Function]}
        >
          <span
            className="martini-tree-icon martini-jms-queue-icon"
          />
          queue://nice
        </li>
      </ul>
    </div>
  </div>
</form>
`;

exports[`DestinationPicker should be rendered with default selected destination 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
  style={
    Object {
      "display": "grid",
      "gridColumnGap": 10,
      "gridTemplateColumns": "max-content 1fr",
    }
  }
>
  <label
    htmlFor="create"
    onContextMenu={[Function]}
    style={
      Object {
        "marginTop": "3px",
        "verticalAlign": "center",
      }
    }
  >
    From:
  </label>
  <div
    style={
      Object {
        "display": "grid",
        "gridTemplateColumns": "1fr",
        "gridTemplateRows": "max-content 1fr",
      }
    }
  >
    <div
      className="sc-AxirZ fiReNW"
    >
      <input
        checked={true}
        name="create"
        onChange={[Function]}
        type="radio"
        value={false}
      />
      <div
        className="label"
        onClick={[Function]}
      >
        Existing JMS Queues and Topics
      </div>
    </div>
    <div
      className="sc-AxirZ fiReNW"
    >
      <input
        checked={false}
        name="create"
        onChange={[Function]}
        type="radio"
        value={true}
      />
      <div
        className="label"
        onClick={[Function]}
      >
        New JMS Queue or Topic
      </div>
    </div>
    <div
      className="form-error-message-wrapper"
    />
  </div>
  <div
    style={
      Object {
        "gridColumn": "span 2",
        "height": "310px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "grid",
          "gridRowGap": "4px",
          "gridTemplateColumns": "1fr",
          "gridTemplateRows": "max-content 1fr",
          "height": "100%",
        }
      }
    >
      <input
        className="theia-input"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search..."
        type="text"
      />
      <ul
        className="sc-AxjAm hGTuJe theia-input"
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <li
          className="item "
          onClick={[Function]}
          onDoubleClick={[Function]}
        >
          <span
            className="martini-tree-icon martini-jms-queue-icon"
          />
          queue://helloWorld
        </li>
        <li
          className="item item-selected"
          onClick={[Function]}
          onDoubleClick={[Function]}
        >
          <span
            className="martini-tree-icon martini-jms-topic-icon"
          />
          topic://testing
        </li>
        <li
          className="item "
          onClick={[Function]}
          onDoubleClick={[Function]}
        >
          <span
            className="martini-tree-icon martini-jms-queue-icon"
          />
          queue://nice
        </li>
      </ul>
    </div>
  </div>
</form>
`;
